### PR TITLE
Issue515 domain decompose refactor

### LINKED ISSUE #515: Move logic related to partition file  to ModelMetadata
Currently reading and setting metadata related to MPI domain decomposition is done inside RectGrid classes. However the same functionality will be also needed in order to parallelise   ParaGrid . So maybe the relevant code should be moved to ModelMetadata classes so it can be reused.

### PR DESCRIPTION
**Pull request by draenog**
# Move domain decompose code to ModelMetadata

### CODE REVIEW

**Comment by draenog** [comment]
It should go on top of #514.

**Review by MarionBWeinzierl — APPROVED** [review]
This looks relatively straightforward to me: the mpi code is moved to ModelMetadata, and the partition file is removed from the function calls.

File: core/src/Model.cpp [hunk 1]
```diff
@@ -119,11 +119,11 @@ void Model::configure()
 #ifdef USE_MPI
     std::string partitionFile
         = Configured::getConfiguration(keyMap.at(PARTITIONFILE_KEY), std::string("partition.nc"));
+    m_etadata.getPartitionMetadata(partitionFile);
```
Comment (TomMelt) [inline core/src/Model.cpp:122]: it's really not a big issue, but I am just curious. Why is the metadata called `m_etadata`?

**Review by TomMelt — APPROVED** [review]
Looks good to me.

File: core/src/Model.cpp (hunk 1 above)
Comment (draenog) [reply core/src/Model.cpp:122]: This is question to @timspainNERSC.

Comment (timspainNERSC) [reply core/src/Model.cpp:122]: <_< >_> 
No good reason.
Some classes, especially from early on in development, have a `m_` prefix for member variables. It slightly amused me to not repeat the 'm' in `m_metadata`.

Comment (TomMelt) [reply core/src/Model.cpp:122]: fair enough. Thanks for the answer.

**Pull request merged**